metrics/histogram.go

```diff
@@ -0,0 +1,112 @@
+package wavefront
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/wavefronthq/wavefront-sdk-go/senders"
+
+	tdigest "github.com/caio/go-tdigest"
+)
+
+type Histogram struct {
+	mutex              sync.Mutex
+	priorTimedBinsList []*timedBin
+	currentTimedBin    *timedBin
+
+	metricName string
```

Comment:
This doesn't belong here. Let's remove the metricName, source and tags from this. This should be purely about creating and updating a histogram. The reporting part such as name etc can be handled in user code.